packages/core/src/external/mpi/normalize-patient.ts

```diff
@@ -0,0 +1,165 @@
+import { PatientDataMPI, Address } from "./patient";
+
+// Define default values for each field
+const defaultValues = {
+  firstName: "john",
+  lastName: "doe",
+  address: [{ addressLine1: "123 main street", city: "anytown", zip: "00000" }],
+  contact: [{ email: "example@example.com", phone: "0000000000" }],
+};
+
+/**
+ * The function checks if a patient's address or name matches the default values and returns null if
+ * they do, otherwise it updates the patient's contact information and returns the modified patient
+ * object.
+ * @param patient - The `patient` parameter is an object of type `Patient`. It
+ * represents the data of a patient, including their address, name, and contact information.
+ * @returns either a modified `Patient` object or `null`.
+ */
+function handleDefaultValues(patient: PatientDataMPI): PatientDataMPI | null {
+  const isDefaultAddress = patient.address?.some(
+    addr =>
+      addr &&
+      (addr.addressLine1 === defaultValues.address?.[0]?.addressLine1 ||
+        addr.city === defaultValues.address?.[0]?.city ||
+        addr.zip === defaultValues.address?.[0]?.zip)
+  );
+
+  const isDefaultName =
+    patient.firstName === defaultValues.firstName && patient.lastName === defaultValues.lastName;
+
+  if (isDefaultAddress || isDefaultName) {
+    return null;
+  }
+
+  patient.contact = (patient.contact ?? []).map(contact => {
+    const defaultContact = defaultValues.contact?.[0];
+    if (!defaultContact) {
+      return contact;
+    }
+    return {
+      email: contact.email === defaultContact.email ? "" : contact.email,
+      phone: contact.phone === defaultContact.phone ? "" : contact.phone,
+    };
+  });
+
+  return patient;
+}
+
+/**
+ * The function `normalizePatient` takes in patient data and normalizes it by splitting the first
+ * and last names, normalizing email and phone numbers, and formatting the address.
+ * @param patient - The `patient` parameter is an object that represents patient data. It
+ * has the following properties:
+ * @returns a normalized version of the patient data. If the patient data is valid, it will return the
+ * normalized patient data as an object of type `Patient`. If the patient data is null, it will
+ * return null.
+ */
+export function normalizePatientDataMPI(patient: PatientDataMPI): PatientDataMPI | null {
```

Comment:
Why not just `normalizePatient`?

Also, we use `undefined` and not `null` inside the app's code - [link](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#16cd72caa4944a10abfac5bf2d0a13e4).

Lastly, why would a function to normalize data return empty? I think the `handleDefaultValues` function, if kept around, should be used at the validation stage, before the data makes its way inside the application.

Reply:
I am removing the call to handleDefaultValues